### ISSUE TITLE
Authenticating always shows "allow/deny" prompt for legacy linkedin app

### ISSUE THREAD

**Issue by jayswain** [decioferreira/omniauth-linkedin-oauth2#49]
I'm seeing the "Allow/Deny" prompt on the linkedin side, even when I have previously authenticated, and am currently logged into linkedin.

I notice this is happening only with an older linkedin app (4 years old?), and is NOT happening with a freshly created linkedin app.

I've seen similar issues with other oauth2 circles. Google, for example has an option to force the dialog prompt, as well as a way to pass a "allow_prompt=auto" param/value to bypass it if the user has previously authenticated and is logged in.

After looking at the linkedin api info, I haven't come across a param for this.

This issue is more so on Linkedin's side (not this gem), as I'm assuming our older linkedin app is maybe some sort of "legacy" app.

Has anyone dealt with this?

**Comment by frankolson** [comment]
Yeah I am seeing this right now too, but I'm not sure how to fix it yet.

**Comment by frankolson** [comment]
@jayswain, did you ever find any more information about this. I've tried creating a new Linkedin app and switching the client/secret in my code base, but that just invalidates all the previously created identities. I'm guessing oauth uids are not the same across apps.

**Comment by jayswain** [comment]
I have not found more information about this.

Yeah, you can't swap your client/secret without making all of your users re-auth the new application.

I posted the question on Stack overflow as well (as Linkedin suggests), and its been rather quiet:

https://stackoverflow.com/questions/54755421/oauth2-authenticating-always-shows-allow-deny-prompt-for-legacy-linkedin-devel
